### PR TITLE
Fix bug in axes orientation (see #149)

### PR DESCRIPTION
**Pull request by DunklesArchipel**
Checklist
* [x] No news.
* [x] Add test for `axes_orientation`

This is a fix for a bug  (see #149) in axes orientation which would swap x- and y-axis, if the lines connecting the label with the plot have a transform.

I did not add a piece of news since this bug does not exist in v 0.4.0, hence it is part of the implementation of 0.5.0

@saraedum suggested some code changes in #149, which are unclear to me.

### CODE REVIEW

**Comment by saraedum** [comment]
I think we should add a case that was broken before and works now.

**Comment by saraedum** [comment]
Does this actually fix everything in the website repository? @DunklesArchipel

**Comment by DunklesArchipel** [comment]
Added a test for axes orientation. This will fail for svgdigitizer versions <=0.4.0.
In the SVG file some of the lines of the axes labels have a transform, which would flip the x and y axis. (see #149):

**Comment by saraedum** [comment]
> In the SVG file some of the lines of the axes labels have a transform, which would flip the x and y axis. (see https://github.com/echemdb/svgdigitizer/issues/149):

I can't imagine that this makes any difference. If it does, then there is another problem here.

**Review by saraedum — CHANGES_REQUESTED** [review]
Sorry, this change should not work. My original idea that there was something wrong with the scaling of `µA/cm²` to `A/m²` can't really be true.

**Review by saraedum — APPROVED** [review]
In the example test case you addded, the values on one axis differ by 50 and on the other axis by .8. So there is roughly a factor 100 in difference here.

File: svgdigitizer/svgplot.py [hunk 1]
```diff
@@ -347,10 +347,13 @@ def score(horizontal, vertical):
             from numpy.linalg import qr
 
             A = self._transformation(
-                self.marked_points[f"{horizontal}1"],
-                self.marked_points[f"{horizontal}2"],
-                self.marked_points[f"{vertical}1"],
-                self.marked_points[f"{vertical}2"],
+                # We do not use the actual label values here but pretend that the marks delimit the unit square [0, 1]×[0, 1].
+                # The actual values might be scaled very differently, e.g., one axis being in μA and one axis in V.
+                # This leads to one axis (and its errors) being rescaled very heavily which can lead to problems in the QR decomposition, see #149.
```
Comment (saraedum) [inline svgdigitizer/svgplot.py:352]:
```suggestion
                # This leads to one axis (and its errors) influencing the transformation matrix too much which can lead to problems in the QR decomposition, see #149.
```

**Pull request merged**